### ISSUE TITLE
New cop to replace conditional assignment with or-assignment 

### ISSUE THREAD

**Issue by fatkodima** [rubocop/rubocop#10325]
I've written the following code
```ruby
def lock_retrier=(value)
  if value.nil?
    @lock_retrier = NullLockRetrier.new
  else
    @lock_retrier = value
  end
end
```

But then realized that this should be really written as
```ruby
def lock_retrier=(value)
  @lock_retrier = value || NullLockRetrier.new
end
```

and wondering why rubocop hasn't suggested such change.

Quick search on the rails codebase via regexp revealed this: 
https://github.com/rails/rails/blob/296ef7a17221e81881e38b51aa2b014d7a28bac5/actionpack/lib/action_dispatch/journey/gtg/transition_table.rb#L75-L79

So I suggest implementing a new cop (not sure about its name).

This should also check for cases like
```ruby
if value
  @lock_retrier = value
else
  @lock_retrier =  NullLockRetrier.new
end
```

etc.

**Comment by bbatsov** [comment]
Seems like a good idea to me. I think we have a couple of similar cops already.

**Comment by marcospedro97** [comment]
Can I give it a try? Maybe we can call it something like Style/OrAssignment?

**Comment by fatkodima** [comment]
@marcospedro97 Yes, please, go ahead 👍

**Comment by marcospedro97** [comment]
I've found a cop with this name already, and it's behaviour is too different from the one we're talking about, thinking about naming it Style/SingleLineOrAssginment
